File: adapter/LegacyWeatherService.java
package adapter;

public class LegacyWeatherService {

    private final ThirdPartyWeatherService weatherApi;

    public LegacyWeatherService(ThirdPartyWeatherService weatherApi) {
        this.weatherApi = weatherApi;
    }

    public String getTemperature(String city, String country) {
        return weatherApi.getTemperature(city, country);
    }

}

File: typescript/patterns/adapter/legacy-weather-service.ts
import { ThirdPartyWeatherService } from "./third-party-weather-service";

export class LegacyWeatherService {
  private readonly weatherApi: ThirdPartyWeatherService;

  constructor(weatherApi: ThirdPartyWeatherService) {
    this.weatherApi = weatherApi;
  }

  getTemperature(city: string, country: string): string {
    return this.weatherApi.getTemperature(city, country);
  }
}
